src/main/java/io/camunda/tasklist/CamundaTaskListClient.java

```diff
@@ -91,13 +95,21 @@ public Task claim(String taskId, String assignee, Boolean allowOverrideAssignmen
     }
   }
 
-  public Task completeTask(String taskId, Map<String, Object> variablesMap)
+  public void completeTask(String taskId, Map<String, Object> variablesMap)
       throws TaskListException {
     try {
-      reconnectEventually();
-      List<VariableInputDTO> variables = ConverterUtils.toVariableInput(variablesMap);
-      return ConverterUtils.toTask(
-          taskApi.completeTask(taskId, new TaskCompleteRequest().variables(variables)), null);
+      Task task = getTask(taskId);
+      if (task.getImplementation().equals(Implementation.JOB_WORKER)) {
+        reconnectEventually();
+        List<VariableInputDTO> variables = ConverterUtils.toVariableInput(variablesMap);
+        taskApi.completeTask(taskId, new TaskCompleteRequest().variables(variables));
+      } else if (task.getImplementation().equals(Implementation.ZEEBE_USER_TASK)) {
+        zeebeClient
```

Comment:
Hi @jonathanlukas,

Thanks for your PR, much appreciated. Do you think it could be usefull to test if zeebeClient is null here and in such case throw an exception explaning that "in order to use the zeebe user task implementation through the tasklistClient, you should inject the zeebeClient through the builder".

Reply:
I added assertions on the zeebe client and properties being present